src/3rd_party/CMakeLists.txt

```diff
@@ -84,19 +84,24 @@ if(USE_SENTENCEPIECE)
 
   add_subdirectory(./sentencepiece)
   include_directories(./sentencepiece)
+  if (NOT USE_WASM_COMPATIBLE_SOURCE)
   set_target_properties(spm_encode spm_decode spm_train spm_normalize spm_export_vocab
     PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
+  endif()
+
+  list(APPEND SENTENCEPIECE_TARGETS sentencepiece-static sentencepiece_train-static)
+  if (NOT USE_WASM_COMPATIBLE_SOURCE)
+    list(APPEND SENTENCEPIECE_TARGETS spm_decode spm_encode spm_export_vocab spm_normalize spm_train)
+  endif()
 
   if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 8.0 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.0))
-    foreach(t sentencepiece-static sentencepiece_train-static
```

Comment:
What is the problem here? Didn't we have, sentencepiece binaries require pthread?

Reply:
The change here only disables building sentencepiece executables for wasm builds. For every other configuration, the flow is exactly same as before.